lib/overcommit/hook/pre_commit/case_conflicts.rb

```diff
@@ -0,0 +1,19 @@
+module Overcommit::Hook::PreCommit
+  # Checks for files that would conflict in case-insensitive filesystems
+  # Adapted from https://github.com/pre-commit/pre-commit-hooks
+  class CaseConflicts < Base
+    def run
+      repo_files = Set.new(Overcommit::GitRepo.all_files)
```

Comment:
I'm worried that this will be overly expensive on large repositories. Would it be possible to adjust it to only get files in the same directories as the files that changed?